### DISCUSSION TITLE
Unable to load in ROMS files because lat dimension is the same name as one of its dimensions

### DISCUSSION THREAD

**Discussion by jwongala** [Parcels-code/Parcels#1729] [Q&A]
## Hello! I am trying to load in ROMS files (curvilinear, z layers) and I am getting the below error and I am not sure how to get around this and proceed. Any help is much appreciated! 

Here is a [link](https://drive.google.com/file/d/1vhJK2PAuR05C24ijL8YNbDzV_iSzSlFL/view?usp=sharing) to download the ROMS file I am trying to load in the code. 

### Code

```python
file1 = '/Volumes/HI_ROMS/hioekg/2019/hioekg_forec_20190101.nc'

filenames = {"U": file1,
             "V": file1}

variables = {"U": "u",
             "V": "v"}

dimensions = { "U": {"lon": "lon", "lat": "lat", "depth": "depth", "time": "time"},
              "V": {"lon": "lon", "lat": "lat", "depth": "depth", "time": "time"}}


fieldset = FieldSet.from_nemo(filenames, variables, dimensions, allow_time_extrapolation=True)
```

### Error message
```text
MissingDimensionsError: 'lat' has more than 1-dimension and the same name as one of its dimensions ('lat', 'lon'). xarray disallows such variables because they conflict with the coordinates used to label dimensions.
```

**Comment by erikvansebille** [comment]
Dear @jwongala, what you report is an `xarray` error, not a Parcels error. So I guess you should check with them how to fix this?

**Comment by JamiePringle** [comment]
Dear @jwongala -- would you mind posting the "ncdump -h hioekg_forec_20190101.nc" output from your ROMS files?  I would not expect to  find variables or dimensions with names like "lon" or "lat" in ROMS output. Since it is on a C-grid, the names should be in the form of "eta_rho" and "xi_rho" or "lon_rho" and "lat_rho", and there should be one set of grids for each of rho, u, and v. 

Also, are you using some custom variant of ROMS? Or have you regridded the data onto an A-grid?  This will cause many problems. 

Jamie

**Reply by jwongala** [reply]
Hi @JamiePringle, below is the ncdump output. The ROMS I am using was provided to me like this. It was originally on a hybrid depth-coordinate and is then regridded to the format I have now. 

```
dimensions:
	lat = 90 ;
	lon = 190 ;
	depth = 18 ;
	time = UNLIMITED ; // (145 currently)
variables:
	float lat(lat, lon) ;
		lat:long_name = "Latitude" ;
		lat:units = "degrees_north" ;
	float lon(lat, lon) ;
		lon:long_name = "Longitude" ;
		lon:units = "degrees_east" ;
	float depth(depth) ;
		depth:long_name = "Z-Level Depth" ;
		depth:units = "meter" ;
		depth:positive = "down" ;
	short mask(lat, lon) ;
		mask:long_name = "field mask" ;
		mask:value = "0-land, 1-water" ;
	float time(time) ;
		time:long_name = "time since initialization" ;
		time:units = "hours since 2000-01-01 00:00:00" ;
	float zeta(time, lat, lon) ;
		zeta:long_name = "free-surface" ;
		zeta:units = "meter" ;
		zeta:coordinates = "lon lat" ;
		zeta:missing_value = 9.99e+10f ;
	float u(time, depth, lat, lon) ;
		u:long_name = "u-momentum component" ;
		u:units = "meter second-1" ;
		u:coordinates = "lon lat" ;
		u:missing_value = 9.99e+10f ;
	float v(time, depth, lat, lon) ;
		v:long_name = "v-momentum component" ;
		v:units = "meter second-1" ;
		v:coordinates = "lon lat" ;
		v:missing_value = 9.99e+10f ;
	float temp(time, depth, lat, lon) ;
		temp:long_name = "potential temperature" ;
		temp:units = "Celsius" ;
		temp:coordinates = "lon lat" ;
		temp:missing_value = 9.99e+10f ;
	float salt(time, depth, lat, lon) ;
		salt:long_name = "salinity" ;
		salt:missing_value = 9.99e+10f ;
		salt:coordinates = "lon lat" ;

// global attributes:
		:author = "ocean" ;
		:history = "Created on Tue, January 01, 2019 at 11:18" ;
		:title = "Zlevel Model Data" ;
}
```

**Comment by JamiePringle** [comment]
This has been regrided onto an A-Grid, where u and v velocities have been  put on the same coordinate grid. You should not use the c-grid formulations like NEMO, but you should instead look at one of the examples where the velocities are all at the same location. 

Note that you only have the horizontal velocities, which will limit you to tracking horizontal motions. 

This does not solve the source of your original error message, which has to do with the fact that "lon" and "lat" are both the name of a dimension and the name of a variable. You can fix this by following the advice in https://stackoverflow.com/questions/55109481/how-to-import-netcdf4-file-with-xarray-when-index-names-have-multiple-dimensions
